wcomponents-core/src/main/java/com/github/bordertech/wcomponents/AbstractWComponent.java

```diff
@@ -436,11 +437,11 @@ public final void serviceRequest(final Request request) {
 	 */
 	private static void collateVisible(final WComponent component, final List<WComponent> list) {
 		if (component.isVisible()) {
+
 			if (component instanceof Container) {
-				final int size = ((Container) component).getChildCount();
 
-				for (int i = 0; i < size; i++) {
-					collateVisible(((Container) component).getChildAt(i), list);
+				for (WComponent child : ((Container) component).getChildren()) {

```

Comment:
The new getChildren() method is looking really good and makes the API easier to use.

As every component extends AbstractWComponent, we have to be careful of performance.

The new getChildren() method needs to be looked at as a "public" API - its job is to make writing code easier for application developers. So it does make sense it returns the list of children as an unmodifiable list to protect internal data from external code.

But when AbstractComponent needs to work internally with the list of children, it does not need to have the overhead of creating an unmodifiable list - we are not trying to make life easier for framework developers.

So for performance, it would be better that AbstractWComponent continue to use the list directly off the model.

Even though we don't get to use a nice "for each" pattern internally it is better for performance as we don't create the unmodifiable list which only exists to protect internal data from external code.